activation/poet.go

```diff
@@ -20,10 +20,21 @@ import (
 )
 
 var (
-	ErrNotFound    = errors.New("not found")
-	ErrUnavailable = errors.New("unavailable")
+	ErrNotFound       = errors.New("not found")
+	ErrUnavailable    = errors.New("unavailable")
+	ErrInvalidRequest = errors.New("invalid request")
 )
 
+type PoetPowParams struct {
+	Challenge  []byte
+	Difficulty uint
```

Comment:
i thought Challenge can be integer, something changed?

you don't want to put precise size for Difficulty? even if you know that it always fits into u32, it would be cleaner to have size

Reply:
The challenge could be an integer. I used the root of the previous poet Merkle tree as the challenge so it was natural to pick a byte slice for it.

Regarding the type of Difficulty, it's the number of leading zero bits the hash output must have at least. I prefer to use normal (u)int in this case, similarly to what functions len() in Go return. (On a side note, I don't understand why they return signed int 🤷).